admin/javascript/src/components/text-field/index.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import SilverStripeComponent from 'silverstripe-component';
+
+class TextFieldComponent extends SilverStripeComponent {
+
+  constructor(props) {
+    super(props);
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  render() {
+    return (
+      <div className="field text">
+        {this.props.label &&
+          <label className="left" htmlFor={`gallery_${this.props.name}`}>
+            {this.props.label}
+          </label>
+        }
+        <div className="middleColumn">
+          <input {...this.getInputProps()} />
```

Comment:
Same comment as for HiddenField, seems over-engineered to have this split out. `{...props}` seems very idiomatic (I see unclecheese is basically saying the same thing :)).